test/threads.jl

```diff
@@ -141,3 +141,91 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic memory ordering
+type CommBuf{T}
+    var1::Ref{T}
```

Comment:
The correctness of the test doesn't and shouldn't depend on the cache layout so this is not necessary.

Also this doesn't ensure they are in different cache, since both of them are likely be allocated in neighboring cells in the pool. The right way to ensure they are in different cache is to add padding for non-atomic members, for atomic members, there's currently no way to do that.